lib/Doctrine/DBAL/Types/ObjectType.php

```diff
@@ -58,10 +58,13 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
         }
 
         $value = (is_resource($value)) ? stream_get_contents($value) : $value;
-        $val   = @unserialize($value);
-        if ($val === false && $value !== 'b:0;') {
-            throw ConversionException::conversionFailed($value, $this->getName());
-        }
+
+        set_error_handler(function ($errno, $errstr) use ($value) {
+            throw ConversionException::conversionFailedUnserialization($value, $this->getName(), $errstr);
+        });
+
+        $val = unserialize($value);
```

Comment:
In the case of an exception, the error handler should be restored as well. Please use `try/finally`.